packages/components/src/components/tabs/tabs.js

```diff
@@ -114,21 +114,40 @@ class Tab extends ContentSwitcher {
   _handleKeyDown(event) {
     const triggerNode = eventMatches(event, this.options.selectorTrigger);
     if (triggerNode) {
+      // enter
       if (event.which === 13) {
         this._updateMenuState();
       }
       return;
     }
 
+    const buttons = toArray(
+      this.element.querySelectorAll(this.options.selectorButtonEnabled)
+    );
+
+    // End or Home key handler
+    if (event.which === 35 || event.which === 36) {
+      event.preventDefault();
+      const activeIndex = {
+        35: buttons.length - 1,
+        36: 0,
+      }[event.which];
```

Comment:
Instead of relying on `event.which` should we abstract into a `match` helper? e.g.

```js
let activeIndex = null

if (match(event, keys.End)) {
  activeIndex = buttons.length - 1;
}

if (match(event, keys.Home) {
  activeIndex = 0;
}
```

Reply:
do you mean I should directly import the match function from the react package or implement it in the vanilla package as well?

Reply:
Oh duh, sorry totally didn't see this was in vanilla